src/JsonRpcRequest.ts

```diff
@@ -0,0 +1,19 @@
+import { JsonStruct } from '@metamask/utils';
+import type { Infer } from 'superstruct';
+import { array, literal, number, record, string, union } from 'superstruct';
+
+import { exactOptional, object } from './superstruct';
+
+export const JsonRpcRequestStruct = object({
```

Comment:
Isn't it already defined in `@metamaks/utils`?

Reply:
Yup as discussed together, I mostly imported the code coming from the `keyring-api`, but the ones coming from `@metamask/utils` seems similar, so lemme use those yes!